src/nanoarrow/nanoarrow_device_cuda.c

```diff
@@ -400,6 +670,58 @@ static void ArrowDeviceCudaRelease(struct ArrowDevice* device) {
   device->release = NULL;
 }
 
+static ArrowErrorCode ArrowDeviceCudaArrayViewCopy(struct ArrowDeviceArrayView* src,
+                                                   struct ArrowDevice* device_dst,
+                                                   struct ArrowDeviceArray* dst) {
+  // For CUDA_HOST, the default implementation is sufficient
+  // TODO: Make sure we've synchronized here when CUDA_HOST is the source.
+  // cuCtxSynchronize() is probably necessary there (since the implementation
+  // calls raw memcpys)
```

Comment:
Should only need to synchronize the relevant stream still. `cuCtxSynchronize()` synchronizes the entire device I believe.